src/lib/types/experimental.ts

```diff
@@ -0,0 +1,31 @@
+export interface IFlags {
+    [key: string]: boolean;
+}
+export interface IExperimentalOptions {
+    flags?: {
+        [key: string]: boolean;
+    };
+    anonymiseEventLog?: boolean;
+    userGroups?: boolean;
+    embedProxy?: boolean;
+    batchMetrics?: boolean;
+    dynamicFlags?: string[];
+    externalResolver?: IExternalFlagResolver;
+}
+
+export interface IUIFlags extends IFlags {
+    ENABLE_DARK_MODE_SUPPORT?: boolean;
+}
+
+export interface IFlagContext {
+    [key: string]: string;
+}
+
+export interface IFlagResolver {
```

Comment:
Could consider merging UI flags and experimental flags into one flat map of flags. 

Not sure if the distinction between an experimental and an UI flag is worth keeping.

Reply:
we could. I was actually thinking it would just enrich the ui-flags with experimental flags, makes sense to have them available in the UI as well. What do you think?

Reply:
Yeah, it would be really nice to automatically have new experimental flags available on the frontend. :+1: 

I was thinking we could also remove the distinction on the backend (and just have "flags"), but not sure.